pkg/provider/pulumi/aws/gateway.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/apigatewayv2"
+	awslambda "github.com/pulumi/pulumi-aws/sdk/v4/go/aws/lambda"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type ApiGatewayArgs struct {
+	ApiFilePath     string
+	LambdaFunctions map[string]*Lambda
+}
+
+type ApiGateway struct {
+	pulumi.ResourceState
+
+	Name string
+	Api  *apigatewayv2.Api
+}
+
+func newApiGateway(ctx *pulumi.Context, name string, args *ApiGatewayArgs, opts ...pulumi.ResourceOption) (*ApiGateway, error) {
+	res := &ApiGateway{Name: name}
+	err := ctx.RegisterComponentResource("nitric:api:AwsApiGateway", name, res, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := openapi3.NewLoader().LoadFromFile(args.ApiFilePath)
```

Comment:
Forgot about this, this should already be loaded as part of the stack already and shouldn't need to be done again. Current this field is unexported `stack.apiDocs`. We'll probably want to read from there/export the field or include a method to read it. This will ensure that api deployments can work from exclusively in memory open api specs.